src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -30,12 +30,9 @@ public function onKernelRequest(GetResponseEvent $event)
         }
 
         $request = $event->getRequest();
-        $session = $this->getSession();
-        if (null === $session || $request->hasSession()) {
-            return;
+        if (!$request->hasSession()) {
+            $request->setSessionFactory(function () { return $this->getSession(); });
```

Comment:
Wouldn't it be more secure to isolate the listener in the closure by injecting it via a `use` instead of using `$this` that can be changed before executing the closure? Maybe I'm being paranoid, so just asking 😄

Reply:
Wow we never closed that far. Not sure it makes sense.
Here particularly there is nothing to close: the closure cannot be accessed.